draft-irtf-cfrg-bls-signature.md

```diff
@@ -1083,6 +1102,114 @@ Procedure:
+At a high level, the signer and verifier interact to compute a blind signature
+protocol yielding `signature = Sign(SK, message)`, where `msg` is the private
+message to be signed,  and `SK` is the server's private key. In this protocol,
+the signer learns nothing of `message`, whereas the verifier learns `signature`
+and nothing of `SK`.
+
+This protocol roughly runs as follows:
+
+~~~
+   Verifier(PK, message)                       Signer(SK)
+  -------------------------------------------------------
+   blind_msg, R = Blind(PK, message)
+
+                         blind_msg
+                        ---------->
+
+                     blind_sig = BlindSign(SK, blind_msg)
+
+                         blind_sig
+                        <----------
+
+  signature = Finalize(PK, message, R, blind_sig)
+~~~
+
+Upon completion, correctness requires that clients can verify signature `signature`
+over private input message `message` using the server public key `PK` by invoking
+the Verify routine. The finalization function performs that check before returning
+the signature.
+
+## Blind {#proto-blind}
+
+The Blind function takes as input the signer public key `PK` and message to sign
+`message` and produces a blinded message and blind inverse.
+
+~~~
+signature = Blind(PK, message)
+
+Inputs:
+- PK, a public key in the format output by SkToPk.
+- message, an octet string.
+
+Outputs:
+- R, a scalar in G1.
+- blinded_msg, an octet string.
+
+Procedure:
+1. R = G1.RandomScalar()
+2. T = hash_to_point(PK || message)
+3. blinded_msg = point_to_signature(R * T)
+4. return R, blinded_msg
+~~~
+
+## BlindSign {#proto-sign}
+
+The BlindSign function invokes CoreEvaluate directly without modification.
+Signers SHOULD perform a membership check on the input blind message Q before
+the result is passed to CoreEvaluate. 
+
+<!-- What is the best way to invoke a membership test here? -->
+
+~~~
+signature = BlindSign(blinded_msg)
+
+Inputs:
+- blind_message, an octet string.
+
+Outputs:
+- blind_sig, an octet string.
+
+Procedure:
+1. Q = signature_to_point(blinded_message)
+2. Z = CoreEvaluate(SK, Q)
+3. blind_sig = point_to_signature(Z)
+4. return blind_sig
+~~~
+
+## Finalize {#proto-finalize}
+
+The Finalize function unblinds the signer's blind signature, verifies
+the corresponding signature, and outputs the resulting signature if valid.
+Otherwise, it raises an error. Verifiers SHOULD perform a membership check 
+on the input blind signature Y before the result is combined with the blind
+inverse (R\_inv). 
+
+~~~
+signature = Finalize(message, PK, R, blind_sig)
+
+Inputs:
+- message, an octet string.
+- PK, a public key in the format output by SkToPk.
+- R, blind inverse output from Blind.
+- blind_sig, an octet string.
+
+Outputs:
+- blind_sig, an octet string.
+
+Procedure:
+1. Y = signature_to_point(blind_sig)
+2. R_inv = R^-1
+3. signature = point_to_signature(R_inv * Y)
+4. if CoreVerify(PK, message, signature), return signature
```

Comment:
Well, not `message`, but `PK || message`, right? I mean since you blind `PK || message`, you have to verify `PK || message`...

Reply:
On tangential note. `PK || message` spells message-augmentation and question if it should be reflected in DST. Well, question is also "what would be the [recommended] DST."

Reply:
Oops, good catch! I was questioning why'd even both with the basic mode while writing this. Is there a benefit to not always doing augmentation?

Reply:
> Is there a benefit to not always doing augmentation?

Do you mean like "ever"? I.e. not only here, but everywhere else and always?

Reply:
> it's a separate question

I'm not very good at reading cues, is it "it was a distraction, forget about it," or "any thoughts?"

Reply:
Oh, sorry :) I mean, what do you think about dropping basic mode altogether and _always_ augmenting the message? (We might want to just have this discussion on the list?)

Reply:
> what do you think about dropping basic mode altogether and always augmenting the message?

I can't think it. Well, I can think of these two things separately but not as one or two interdependent. I mean one can discuss omission of the Basic scheme, but it doesn't actually translate to "always augment." Because at least you need a non-augmented sign for PoP. Or shall we discuss removing even PoP? But then this asks for the question "what's criteria for inclusion/omission"? Note that this can be turned even against this PR. Indeed, why **this** blind signature scheme? ;-)

As for the referred message. In my mind it effectively asks why doesn't the draft describe a specific application? Well, I can't speak for the draft authors, but it appears to me that it was the original intent. That is to be application-agnostic, and rather provide a kind of a foundation for specific applications. I mean the correct question would be rather "where are the application specifications?" and not "why doesn't **this** draft describe one/mine?" I might be wrong of course, as, again, I don't claim to know the original intent or its evolution :-)